packages/presets/src/ai/ai-spec.ts

```diff
@@ -0,0 +1,126 @@
+import type { BlockSpec } from '@blocksuite/block-std';
+import {
+  AFFINE_AI_PANEL_WIDGET,
+  AFFINE_EDGELESS_COPILOT_WIDGET,
+  AffineAIPanelWidget,
+  AffineFormatBarWidget,
+  AffineSlashMenuWidget,
+  EdgelessCopilotWidget,
+  EdgelessElementToolbarWidget,
+  EdgelessRootBlockSpec,
+  PageRootBlockSpec,
+  ParagraphBlockService,
+  ParagraphBlockSpec,
+} from '@blocksuite/blocks';
+import { assertInstanceOf } from '@blocksuite/global/utils';
+import { literal, unsafeStatic } from 'lit/static-html.js';
+
+import { buildAIPanelConfig } from './ai-panel.js';
+import {
+  setupEdgelessCopilot,
+  setupEdgelessElementToolbarEntry,
+} from './entries/edgeless/index.js';
+import { setupFormatBarEntry } from './entries/format-bar/setup-format-bar.js';
+import { setupSlashMenuEntry } from './entries/slash-menu/setup-slash-menu.js';
+import { setupSpaceEntry } from './entries/space/setup-space.js';
+
+export const AIPageRootBlockSpec: BlockSpec = {
```

Comment:
I still think it is better to use patch function instead. Imagine we have different types of provided specs, then there is no way to use them together in this way.

Reply:
Reasonable, but I will change the patch function to receive a specific single block spec instead of an spec array, because now we encourage AFFiNE to assemble the required block specs themselves.

Reply:
I'm not so sure what if certain feature like AI requires patching multiple specs. Individual patch function may not work at all on its own, unless different specs are patched all together.